wherobots/db/connection.py

```diff
@@ -178,6 +178,8 @@ def _handle_results(self, execution_id: str, results: Dict[str, Any]) -> Any:
         if result_format == ResultsFormat.JSON:
             return json.loads(result_bytes.decode("utf-8"))
         elif result_format == ResultsFormat.ARROW:
+            import pyarrow
```

Comment:
I understand it's nice(r) for this import to be done only if we're actually dealing with Arrow results, but I think we should make (or at least try/verify) this import at the time of the _request_ in `__request_results()` when the requested results format in Arrow.